hacks/BusPirateSetup.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/sh
+# BusPirateSetup
+# This script generates the file $HOME/.config/buspirate
+# It lets you configure
+#	1) If you have a BusPirate 5XL instead of a BusPirate 6
+#	2) If you have a BusPirate 5 Rev 8 instead of the released BusPirate 5
+#	3) The directory where new versions of the firmware are located
+#   4) Which terminal emulation program you use
+
+CONFIG=config
+# You can re-run this program at any time to modify these values
+# Optionally, you can edit ~/.config/buspirate/$CONFIG
+
+# Time-stamp: <2024-12-04 09:18:21 (grymoire)>
+
+# make sure there is a config file is there first
+if [ ! -d "$HOME/.config" ]
+then
+    mkdir "$HOME/.config"
+fi
+if [ ! -d "$HOME/.config/buspirate" ]
+then
+    mkdir "$HOME/.config/buspirate"
+fi
+if [ ! -f "$HOME/.config/buspirate/${CONFIG}" ]
+then
+   touch "$HOME/.config/buspirate/${CONFIG}"
+fi
+   
+   BP8=0
+   printf "%s\n" "--------------------"
+   printf "%s\n" "Selecting the default firmware for RP2040-based systems"
```

Comment:
Well, on my system, dmesg  and/or reading syslog requires root permission. And I'm not assuming udev rules have been installed. I targeted people who have little experience in setting up the BP. I even tried to pick good default terminal emulation options. I didn't want to have a script perform a sudo because of this. But perhaps there is another way? Ideas anyone?